syncer/filter.go

```diff
@@ -29,26 +29,27 @@ import (
 // - any schema of table names is system schema.
 // - any table name doesn't pass block-allow list.
 // - type of SQL doesn't pass binlog filter.
-func (s *Syncer) skipQuery(tables []*filter.Table, stmt ast.StmtNode, sql string) (bool, error) {
+// skipQuery return (ignored, isBaList, error).
+func (s *Syncer) skipQuery(tables []*filter.Table, stmt ast.StmtNode, sql string) (bool, bool, error) {
 	if utils.IsBuildInSkipDDL(sql) {
-		return true, nil
+		return true, false, nil
```

Comment:
because we will `trackDDL` for false isBaList, return false here will cause schema tracker execute a not supported DDL.

I recommend change `isBaList` to `needTrack` and adjust the logic of returning value.